File: components/ng2-bootstrap-config.ts
import {Type} from '@angular/core';

export enum Ng2BootstrapTheme {BS3 = 1, BS4 = 2}

export class Ng2BootstrapConfig {
  private static _theme:Ng2BootstrapTheme;

  public static get theme():Ng2BootstrapTheme {
    // hack as for now
    if (Type && (Type as any).__theme === 'bs4') {
      return Ng2BootstrapTheme.BS4;
    }
    return (this._theme || Ng2BootstrapTheme.BS3);
  }

  public static set theme(v:Ng2BootstrapTheme) {
    this._theme = v;
  }
}

File: lib/ng2-bootstrap-config.dart
import 'dart:html';
import 'package:node_shims/js.dart';

enum Ng2BootstrapTheme { BS3, BS4 }

class Ng2BootstrapConfig {
  static Ng2BootstrapTheme _theme;

  static Ng2BootstrapTheme get theme {
    // hack as for now
    dynamic w = window;
    if (w && identical(w.___theme, "bs4")) {
      return Ng2BootstrapTheme.BS4;
    }
    return or(Ng2BootstrapConfig._theme, Ng2BootstrapTheme.BS3);
  }

  static set theme(Ng2BootstrapTheme v) {
    Ng2BootstrapConfig._theme = v;
  }
}